File: src/Level/Rating/power_rating.py
from rating import Rating

class PowerRating(Rating):
    """ Represents the Power Rating """
    
    def __init__(self, power, level):
        """ Initialize the Power Rating """
        Rating.__init__(self, level)
        self.power = power
    
    def usePower(self, amount):
        """ Increment the move counter by 1 """
        if not self.level.finished():
            self.power -= amount
        
    def checkAwarded(self):
        """ Check if the Rating should be awarded """
        self.awarded = self.power >= 15

File: src/android/src/com/kaottessur/pymine/level/rating/PowerRating.java
package com.kaottessur.pymine.level.rating;

import com.kaottessur.pymine.level.Level;

public class PowerRating extends Rating {
	private int power;
	
	public PowerRating(Level level, int power) {
		super(level);
		this.power = power;
	}
	
	public void usePower(int amount) {
		if (!level.finished())
			power -= amount;
	}
	
	public int getPower() {
		return power;
	}

	@Override
	public void checkAwarded() {
		awarded = (power >= 15);
	}

}
